core/domain/interaction_jobs_one_off_test.py

```diff
@@ -1356,3 +1356,112 @@ def test_no_action_is_performed_for_deleted_exploration(self):
                 interaction_jobs_one_off
                 .RuleInputToCustomizationArgsMappingOneOffJob)
         )
+
+
+class LogicProofInteractionOneOffJobTests(test_utils.GenericTestBase):
+
+    ALBERT_EMAIL = 'albert@example.com'
+    ALBERT_NAME = 'albert'
+
+    VALID_EXP_ID = 'exp_id0'
+    NEW_EXP_ID = 'exp_id1'
+    EXP_TITLE = 'title'
+
+    def setUp(self):
+        super(LogicProofInteractionOneOffJobTests, self).setUp()
+
+        # Setup user who will own the test explorations.
+        self.signup(self.ALBERT_EMAIL, self.ALBERT_NAME)
+        self.albert_id = self.get_user_id_from_email(self.ALBERT_EMAIL)
+        self.process_and_flush_pending_mapreduce_tasks()
+
+    def test_exp_state_pairs_are_produced_only_for_desired_interactions(self):
+        """Checks output pairs are produced only for
+        desired interactions.
+        """
+        exploration = exp_domain.Exploration.create_default_exploration(
+            self.VALID_EXP_ID, title='title', category='category')
+
+        exploration.add_states(['State1', 'State2'])
+
+        state1 = exploration.states['State1']
+
+        state1.update_interaction_id('LogicProof')
+
+        customization_args_dict1 = {
+            'question': {'value': {
```

Comment:
Nit pick: please break this onto its own line.

Reply:
done